nuxeo-management-rest-api/src/test/java/org/nuxeo/rest/management/TestElasticsearchObject.java

```diff
@@ -0,0 +1,126 @@
+import org.nuxeo.runtime.test.runner.Features;
+import org.nuxeo.runtime.test.runner.FeaturesRunner;
+import org.nuxeo.runtime.transaction.TransactionHelper;
+
+/**
+ * @since 11.1
+ */
+@RunWith(FeaturesRunner.class)
+@Features(AutomationFeature.class)
+@Deploy("org.nuxeo.ecm.core.management")
+@Deploy("org.nuxeo.rest.management.test")
+@Deploy("org.nuxeo.ecm.automation.core")
+@Deploy("org.nuxeo.elasticsearch.core")
+@Deploy("org.nuxeo.elasticsearch.automation")
+public class TestElasticsearchObject extends ManagementBaseTest {
+
+    public static final String GET_DOCUMENTS_QUERY = "SELECT * from Document";
+
+    @Inject
+    protected CoreSession coreSession;
+
+    @Inject
+    protected ElasticSearchService ess;
+
+    @Inject
+    protected ElasticSearchAdmin esa;
+
+    @Test
+    public void shouldRunIndexation() throws Exception {
+        // Init indexes and drop if any
+        esa.initIndexes(true);
+
+        // Initial docs count
+        long initialDocCount = coreSession.query(GET_DOCUMENTS_QUERY).totalSize();
+
+        // Nothing indexed because of disable indexing flag
+        assertEquals(0, ess.query(new NxQueryBuilder(coreSession).nxql(GET_DOCUMENTS_QUERY)).totalSize());
+
+        // Creates a new docs without indexing them
+        createDocuments();
+
+        // Wait for an eventual indexation
+        TransactionHelper.startTransaction();
+        waitForIndexing();
+        assertEquals(0, ess.query(new NxQueryBuilder(coreSession).nxql(GET_DOCUMENTS_QUERY)).totalSize());
+
+        // Starts the ES indexing
+        String path = String.format("site/management/elasticsearch/%s/reindex", coreSession.getRepositoryName());
+        try (CloseableClientResponse response = httpClientRule.post(path, null)) {
+            assertEquals(SC_NO_CONTENT, response.getStatus());
+            waitForIndexing();
+            assertEquals(2 + initialDocCount,
+                    ess.query(new NxQueryBuilder(coreSession).nxql(GET_DOCUMENTS_QUERY)).totalSize());
+        }
+    }
+
+    @Test
+    public void shouldFailRunningIndexationWhenRepositoryNotExist() {
+        // Launch the ES indexing
+        String path = String.format("site/management/elasticsearch/%s", "unExistingRepository");
+        try (CloseableClientResponse response = httpClientRule.post(path, null)) {
+            assertEquals(SC_NOT_FOUND, response.getStatus());
+        }
+    }
+
+    protected void createDocuments() {
+        Stream.of("my-file-1", "my-file-2").forEach((fileName) -> {
+            DocumentModel doc = coreSession.createDocumentModel("/default-domain/workspaces/", fileName, "File");
+            doc.setPropertyValue("dc:title", String.format("Title of %s", fileName));
+            doc.putContextData(ElasticSearchConstants.DISABLE_AUTO_INDEXING, Boolean.TRUE);
+            coreSession.createDocument(doc);
+        });
+
+        coreSession.save();
+        TransactionHelper.commitOrRollbackTransaction();
```

Comment:
I think that:
- you don't need `coreSession.save()` as you are commitiing the transaction
- you should start a transaction just after committing it to make sure that you always have a transaction in the test
To be checked with @efge.

Reply:
Indeed, see my other comment. In test we should always use the transactional feature.